src/sql/src/plan/func.rs

```diff
@@ -390,14 +390,66 @@ impl ParamList {
     ///
     /// Returns `Some` if the constraints were successfully resolved, or `None`
     /// otherwise.
-    fn resolve_polymorphic_types(&self, typs: &[Option<ScalarType>]) -> Option<ScalarType> {
-        // TODO(sploiselle): support custom types
-        if typs.iter().any(|t| match t {
-            Some(t) => t.is_custom_type(),
-            None => false,
-        }) {
-            return None;
-        }
+    ///
+    /// ## Polymorphism
+    ///
+    /// To handle custom types w/r/t polymorphism, we derive a "complex type
+    /// signature" for the constrained type, which represents the constrained
+    /// type's `custom_oid` and an embdedded element. This complex type
+    /// signature is this function's return value.
+    ///
+    /// We define custom types as any type defined with `CREATE TYPE` or embeds
+    /// a custom type. We'll use the term anonymous type to define all
```

Comment:
Do `anonymous` types include builtin types, as well?

Reply:
`anonymous` types are instances of types that are not `custom` type, so `int`, `int list`, `map[text=>int]`, etc.